src/prpy/tsr/generic.py

```diff
@@ -0,0 +1,275 @@
+import numpy
+from prpy.tsr.tsrlibrary import TSRFactory
+from prpy.tsr.tsr import TSR, TSRChain
+
+def get_manip_idx(robot, manip=None):
+    ''' 
+    Helper function for getting the manipulator index
+    to be used by the TSR
+    '''
+    if manip is None:
+        manip_idx = robot.GetActiveManipulatorIndex()
+    else:
+        import openravepy
+        p = openravepy.KinBody.SaveParameters
+        with robot.CreateRobotStateSaver(p.ActiveManipulator):
+            robot.SetActiveManipulator(manip)
+            manip_idx = manip.GetRobot().GetActiveManipulatorIndex()
+    return manip_idx
+
+def cylinder_grasp(robot, obj, obj_radius, obj_height, 
+                   lateral_offset = 0.0, 
+                   vertical_tolerance = 0.02,
+                   yaw_range = None,
+                   manip = None, **kwargs):
+    """
```

Comment:
Add a one-line description of the function to the docstring. It currently lists assumptions, but does not explain what the function *does*.